openprescribing/pipeline/management/commands/fetch_and_import_ncso_concessions.py

```diff
@@ -223,4 +266,122 @@ def regularise_ncso_name(name):
     # Remove spaces around slashes
     name = re.sub(" */ *", "/", name)
 
+    # dm+d names often have the units appended after the pack size (e.g. "12 tablets"),
+    # we want to remove the units leaving just the "12"
+    name = re.sub(r"(\d+) [^\d]+$", r"\1", name)
```

Comment:
Will this remove some units that we want to keep?  e.g. I can see that it would make sense to remove e.g. "ampoule" at the end of [Adrenaline (base) 1mg/10ml (1 in 10,000) dilute solution for injection ampoules 10 ampoule](https://openprescribing.net/dmd/vmpp/4523811000001103/)

But for e.g. [Adrenaline (base) 1% eye drops 7.5 ml](https://openprescribing.net/dmd/vmpp/3705611000001107/) - it would remove the "ml" - is that what it should do?

(I guess this is only being used to help name-matching, so the question is probably irrelevant)

Reply:
This is a good question and it's prompted me to expand the comments a bit more here. The basic issue is that the names we get from the feed don't have the units, so we have to remove them to in order to match.

The old code effectively did the same thing, but using a substring match:
https://github.com/ebmdatalab/openprescribing/blob/f0ca7a5e41a5aefdbfba125741041c761f35b02f/openprescribing/pipeline/management/commands/fetch_and_import_ncso_concessions.py#L178-L180

There are some rare cases where this results in ambiguous names, which I've documented here:
https://github.com/ebmdatalab/openprescribing/issues/3979

But where there's more than one matching name we now treat that as no match and require a manual resolution:
https://github.com/ebmdatalab/openprescribing/blob/531e0e0a94126f3ae808eceab7b5de4539987bc6/openprescribing/pipeline/management/commands/fetch_and_import_ncso_concessions.py#L205-L207